src/routers/StackRouter.js

```diff
@@ -110,15 +110,25 @@ export default (
           action.type === NavigationActions.NAVIGATE &&
           childRouters[action.routeName] !== undefined
         ) {
+          const childRouter = childRouters[action.routeName];
+          let childState;
+          if (childRouter !== null) {
+            const childAction =
+              action.action ||
+              NavigationActions.init({ params: action.params });
+            childState = childRouter.getStateForAction(childAction);
+          }
+          const route = {
+            ...childState,
+            routeName: action.routeName,
+            key: `Init-${_getUuid()}`,
+          };
+          if (action.params) {
+            route.params = action.params;
+          }
           return {
```

Comment:
Instead of this change, I would favor refactoring this to be similar to [`TabRouter`'s initialization](https://github.com/react-community/react-navigation/blob/master/src/routers/TabRouter.js#L76), where instead of directly returning the initialized state, it replaces the null state in-place and lets the following logic handle the initial action.